src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -257,6 +295,32 @@ class HaAutomationEditor extends LitElement {
     );
   }
 
+  private _toggleYaml(): void {
+    this._showYaml = !this._showYaml;
+  }
+
+  private get _yamlEditor(): HuiYamlEditor | null {
+    return this.shadowRoot!.querySelector("hui-yaml-editor");
+  }
+
+  private get _yaml(): string {
+    if (this._config) {
+      return yaml.safeDump(this._config);
+    }
+    return "";
+  }
+
+  private _handleYamlChanged(ev): boolean {
+    ev.stopPropagation();
+    try {
+      this._config = yaml.safeLoad(ev.detail.value);
```

Comment:
As we don't show a preview, we don't have to update the config on every change of the `yaml`, we can just convert the `yaml` when switching back or when saving. Just save the `yaml` from here in `this._yaml`